src/pgduckdb_ruleutils.cpp

```diff
@@ -635,6 +636,15 @@ pgduckdb_get_tabledef(Oid relation_oid) {
 
 		const char *column_name = NameStr(column->attname);
 
+		/*
+		 * Check that this type is known by DuckDB, and throw the appropriate
+		 * error otherwise. This is particularly important for NUMERIC without
+		 * precision specified. Because that means something very different in
+		 * Postgres
+		 */
+		auto duck_type = pgduckdb::ConvertPostgresToDuckColumnType(column);
+		pgduckdb::GetPostgresDuckDBType(duck_type, true);
```

Comment:
This is a bit too bad but it looks like `GetPostgresDuckDBType` can throw cpp exceptions.. Here it's fine because `pgduckdb_get_tabledef` is only called in a method that is robust to it. But we should finish cleaning up our code everywhere to make sure this is less likely to happen ;-)

Reply:
Changed the new code to throw cpp exceptions for these errors, at least then we're consistent.